src/index.ts

```diff
@@ -1,1314 +1,1236 @@
-   * @default "SECONDARY"
-   */
-  buttonStyle: ButtonStyle;
-  /**
-   * loop through the pages.
-   * @default false
-   */
-  loop: boolean;
-}
+/**
+ * The pagination system
+ */
+export class Pagination extends MessageEmbed {
+	/**
+	 * The interaction that the paginator is for.
+	 * @readonly
+	 */
+	public readonly interaction:
+		| CommandInteraction
+		| MessageComponentInteraction
+		| ContextMenuInteraction
+		| Message;
 
-export interface EmojiOptions {
-  /**
-   * The emoji to use for the first button.
-   * @default "⏮"
-   */
-  firstEmoji: EmojiIdentifierResolvable;
-  /**
-   * The emoji to use for the previous button.
-   * @default "◀️"
-   */
-  prevEmoji: EmojiIdentifierResolvable;
-  /**
-   * The emoji to use for the next button.
-   * @default "▶️"
-   */
-  nextEmoji: EmojiIdentifierResolvable;
-  /**
-   * The emoji to use for the last button.
-   * @default "⏭"
-   */
-  lastEmoji: EmojiIdentifierResolvable;
-}
+	/**
+	 * pagination button infos
+	 * @readonly
+	 * @default {
+	 *  first: {
+	 *    emoji: "⏮",
+	 *    label: "",
+	 *    style: "SECONDARY",
+	 *  },
+	 *  prev: {
+	 *    emoji: "◀️",
+	 *    label: "",
+	 *    style: "SECONDARY",
+	 *  },
+	 *  next: {
+	 *    emoji: "▶️",
+	 *    label: "",
+	 *    style: "SECONDARY",
+	 *  },
+	 *  last: {
+	 *    emoji: "⏭",
+	 *    label: "",
+	 *    style: "SECONDARY",
+	 *  },
+	 * }
+	 */
+	public readonly buttonInfo: ButtonsOptions;
 
-export interface LabelOptions {
-  /**
-   * The label to use for the first button.
-   * @default ""
-   */
-  firstLabel: string;
-  /**
-   * The label to use for the previous button.
-   * @default ""
-   */
-  prevLabel: string;
-  /**
-   * The label to use for the next button.
-   * @default ""
-   */
-  nextLabel: string;
-  /**
-   * The label to use for the last button.
-   * @default ""
-   */
-  lastLabel: string;
-}
+	/**
+	 * the images to paginate through
+	 */
+	public images: string[];
 
-export interface ButtonOptions {
-  /**
-   * The emoji to use for the button.
-   */
-  emoji: EmojiIdentifierResolvable;
-  /**
-   * The text to show on the button.
-   * @default ""
-   */
-  label: string;
-  /**
-   * The style of the button.
-   * @default "SECONDARY"
-   */
-  style: ButtonStyle;
-}
+	/**
+	 * the descriptions to paginate through
+	 */
+	public descriptions: string[];
 
-export interface ButtonsOptions {
-  /**
-   * The first button of the pagination row
-   */
-  first: ButtonOptions;
-  /**
-   * The previous button of the pagination row
-   */
-  prev: ButtonOptions;
-  /**
-   * The next button of the pagination row
-   */
-  next: ButtonOptions;
-  /**
-   * The last button of the pagination row
-   */
-  last: ButtonOptions;
-}
+	/**
+	 * the action rows of the final message
+	 */
+	public actionRows: MessageActionRow[];
 
-/**
- * The style of the paginator buttons.
- */
-export type ButtonStyle = "PRIMARY" | "SECONDARY" | "DANGER" | "SUCCESS";
-class Pagination extends MessageEmbed {
```

Comment:
Your prettier config is a bit different from mine which results is a lots of style changes and a bit hard to review what you changed here. I'll include my prettier config file so that you can use that.